icicle/appUtils/msm/msm.cu

```diff
@@ -144,16 +144,17 @@ __global__ void final_accumulation_kernel(P* final_sums, P* final_results, unsig
 
 //this function computes msm using the bucket method
 template <typename S, typename P, typename A>
-void bucket_method_msm(unsigned bitsize, unsigned c, S *h_scalars, A *h_points, unsigned size, P*h_final_result){
+void bucket_method_msm(unsigned bitsize, unsigned c, S *scalars, A *points, unsigned size, P* final_result, bool on_device) {
```

Comment:
should be separated to input_on_device and output_on_device

Reply:
I kind of assume that everything is on device for simplicity